view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js

```diff
@@ -230,46 +268,72 @@ define(
                      * @returns {boolean}
                      */
                     result.isIdeal = function () {
-                        if (typeof value.type !== 'undefined' && value.type.indexOf("ideal") >= 0) {
+                        if (result.getBrandCode().indexOf("ideal") >= 0) {
                             return true;
                         }
 
                         return false;
                     };
+                    /**
+                     * Checks if payment method is sepa direct debit
+                     */
+                    result.isSepaDirectDebit = function () {
+                        if (result.getBrandCode().indexOf("sepadirectdebit") >= 0) {
+                            return true;
+                        }
 
+                        return false;
+                    };
                     /**
                      * Renders the secure fields,
                      * creates the ideal component,
                      * sets up the callbacks for ideal components and
                      */
                     result.renderIdealComponent = function () {
+                        result.isPlaceOrderAllowed(false);
 
-                        self.isPlaceOrderActionAllowed(false);
+                        var idealNode = document.getElementById('iDealContainer');
 
-                        var secureFieldsNode = document.getElementById('iDealContainer');
+                        var ideal = self.checkoutComponent.create('ideal', {
+                            items: result.getIssuers(),
+                            onChange: function (state) {
+                                if (!!state.isValid) {
```

Comment:
not sure if double "!!" is intentional here

Reply:
@cyattilakiss this should not be needed right ?